### PR TITLE
HPCC-8116 More compiler warning fixes

### PR DESCRIPTION
**Pull request by richardkchapman**
More components can now be built with -Wall set.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

File: rtl/eclrtl/eclrtl.cpp [hunk 1]
```diff
@@ -3085,7 +3012,7 @@ void rtlStrToEStr(unsigned outlen, char *out, unsigned inlen, const char *in)
         out[i] = codepage[j];
     }
     for (;i<outlen; i++)
-        out[i] = codepage[' '];
+        out[i] = codepage[(unsigned) ' '];
```
Comment (richardkchapman) [inline rtl/eclrtl/eclrtl.cpp]: perhaps (unsigned char) would be more correct?

File: system/security/zcrypt/base64.cpp [hunk 2]
```diff
@@ -98,7 +98,7 @@ ZBuffer& base64_decode(int inlen, const char *in, ZBuffer& data)
     {
         for(int i = 0; i < 64; ++i)
         {
-            BASE64_dec[BASE64_enc[i]] = i;
+            BASE64_dec[(int) BASE64_enc[i]] = i;
```
Comment (richardkchapman) [inline system/security/zcrypt/base64.cpp]: Should probably cast to unsigned char instead (won\t cause a problem in practice though)

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: roxie/roxiemem/roxiemem.cpp [hunk 3]
```diff
@@ -1145,8 +1147,8 @@ static void setParallelSyncReleaseGranularity(size_t granularity, unsigned scali
     parallelSyncReleaseThreshold = granularity * scaling;
 }
 #else
-const static size_t parallelSyncReleaseGranularity = DEFAULT_PARALLEL_SYNC_RELEASE_GRANULARITY;
-const static size_t parallelSyncReleaseThreshold = DEFAULT_PARALLEL_SYNC_RELEASE_THRESHOLD;
+// const static size_t parallelSyncReleaseGranularity = DEFAULT_PARALLEL_SYNC_RELEASE_GRANULARITY;
```
Comment (ghalliday) [inline roxie/roxiemem/roxiemem.cpp]: Should be inside #ifdef PARALLEL_SYNC_RELEASE, rather than commented out.

**Comment by ghalliday** [comment]
@richardkchapman one comment

**Comment by richardkchapman** [comment]
@ghalliday Repushed to fix

**Pull request merged**